controller/array_action/array_mediator_svc.py

```diff
@@ -1377,3 +1402,90 @@ def _add_snapshot(self, snapshot_name, source_cli_volume, pool):
+                if NAME_NOT_EXIST_OR_MEET_RULES in ex.my_message:
+                    raise array_errors.InvalidArgumentError(ex.my_message)
+                if NOT_ENOUGH_EXTENTS_IN_POOL_CREATE in ex.my_message:
+                    raise array_errors.NotEnoughSpaceInPool(id_or_name=pool)
+                if any(msg_id in ex.my_message for msg_id in (NON_ASCII_CHARS, INVALID_NAME, TOO_MANY_CHARS)):
+                    raise array_errors.InvalidArgumentError(ex.my_message)
+                raise ex
+        return None
+
+    def _get_volume_id_from_volume_group(self, volume_group_id):
+        filter_value = 'volume_group_id={}'.format(volume_group_id)
+        cli_volume = self._lsvdisk(filtervalue=filter_value)
+        return cli_volume.id
+
+    def _rollback_create_volume_from_snapshot(self, volume_id, volume_group_id):
+        self._delete_volume_by_name(volume_id)
+        self._rmvolumegroup(volume_group_id)
+
+    def _change_volume_group(self, volume_id, volume_group):
+        cli_kwargs = {}
+        if volume_group:
+            cli_kwargs['volumegroup'] = volume_group
+        else:
+            cli_kwargs['novolumegroup'] = True
+        self._chvdisk(volume_id, **cli_kwargs)
+
+    def _rename_volume(self, volume_id, name):
+        self._chvdisk(volume_id, name=name)
+
+    def _chvdisk(self, volume_id, **kwargs):
+        try:
+            self.client.svctask.chvdisk(vdisk_id=volume_id, **kwargs)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if is_warning_message(ex.my_message):
+                logger.warning(
+                    "exception encountered while changing volume parameters '{}': {}".format(kwargs, ex.my_message))
+            else:
+                if OBJ_ALREADY_EXIST in ex.my_message:
+                    raise array_errors.VolumeAlreadyExists(kwargs, self.endpoint)
+                raise ex
+
+    def _rmvolumegroup(self, volume_group_id, not_exist_error=False):
+        logger.info("deleting volume group : {0}".format(volume_group_id))
+        try:
+            self.client.svctask.rmvolumegroup(object_id=volume_group_id)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if is_warning_message(ex.my_message):
+                logger.warning("exception encountered during deletion of volume group {}: {}".format(volume_group_id,
+                                                                                                     ex.my_message))
+            else:
+                logger.error("Failed to delete volume group {}".format(volume_group_id))
+                if OBJ_NOT_FOUND in ex.my_message or VOL_NOT_FOUND in ex.my_message:
+                    logger.warning(array_errors.ObjectNotFoundError(volume_group_id))
+                    if not not_exist_error:
+                        return
+                raise ex
+
+    def _extract_and_remove_volume_group(self, volume_id, volume_group, volume_group_id):
```

Comment:
IMO we don't need this function, since it doesn't do much and we don't have a good name for it

Reply:
done